packages/bson-transpilers/codegeneration/CodeGenerationVisitor.js

```diff
@@ -116,7 +116,12 @@ module.exports = (ANTLRVisitor) => class CodeGenerationVisitor extends ANTLRVisi
         );
       })
       .reduce((obj, c) => {
-        obj[c] = this.Imports[c].template(this.requiredImports[c]);
```

Comment:
Would it make sense to pass both `this.requiredImports[c]` and `mode` as arguments always? That way, all the templates have the same function signature, rather than one being special-cased in a way that’s not obvious just when looking at the YAML files